nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/PutMongoRecord.java

```diff
@@ -131,7 +131,7 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                 for (String name : schema.getFieldNames()) {
```

Comment:
Why not do the conversion while inserting?

Reply:
Mainly preference. When went back and looked at it, a single recursive function seemed like the right call. I still think it is. Not much to be saved by mixing it into that loop IMO.